core/node/libp2p/rcmgr_defaults.go

```diff
@@ -72,33 +72,28 @@ Run 'ipfs swarm limit all' to see the resulting limits.
 			Memory: int64(maxMemory),
 			FD:     int(numFD),
 
-			// By default, we just limit connections on the inbound side.
+			// At least as of 2023-01-25, it's possible to open a connection that 
+			// doesn't for any memory usage with the libp2p Resource Manager/Accountant
+			// (see https://github.com/libp2p/go-libp2p/issues/2010#issuecomment-1404280736).
+			// As a result, we can't curretly rely on Memory limits to full protect us.
+			// Until https://github.com/libp2p/go-libp2p/issues/2010 is addressed, 
+			// we take a proxy now of restricting to 1 inbound connection per MB.
+			// Note: this is more generous than go-libp2p's default autoscaled limits which do
+			// 64 connections per 1GB
+			// (see https://github.com/libp2p/go-libp2p/blob/master/p2p/host/resource-manager/limit_defaults.go#L357 ).
+			ConnsInbound:  maxMemory >> 20, // Steve: I assume this isn't valid Go code.
+
 			Conns:         bigEnough,
-			ConnsInbound:  rcmgr.DefaultLimits.SystemBaseLimit.ConnsInbound, // same as libp2p default
 			ConnsOutbound: bigEnough,
-
-			// We limit streams since they not only take up memory and CPU.
-			// The Memory limit protects us on the memory side,
-			// but a StreamsInbound limit helps protect against unbound CPU consumption from stream processing.
 			Streams:         bigEnough,
-			StreamsInbound:  rcmgr.DefaultLimits.SystemBaseLimit.StreamsInbound,
+			StreamsInbound:  bigEnough,
 			StreamsOutbound: bigEnough,
 		},
-		// Most limits don't see an increase because they're already infinite/bigEnough or at their max value.
-		// The values that should scale based on the amount of memory allocated to libp2p need to increase accordingly.
-		SystemLimitIncrease: rcmgr.BaseLimitIncrease{
-			Memory:     0,
-			FDFraction: 0,
-
-			Conns:         0,
-			ConnsInbound:  rcmgr.DefaultLimits.SystemLimitIncrease.ConnsInbound,
-			ConnsOutbound: 0,
-
-			Streams:         0,
-			StreamsInbound:  rcmgr.DefaultLimits.SystemLimitIncrease.StreamsInbound,
-			StreamsOutbound: 0,
-		},
+		SystemLimitIncrease: noLimitIncrease,
 
+		// Transient connections won't cause any memory to accounted for by the resource manager.
+		// Only established connections do.
+		// As a result, we can't rely on System.Memory to protect us from a bunch of transient connection being opened.
 		TransientBaseLimit: rcmgr.BaseLimit{
```

Comment:
```suggestion
		TransientBaseLimit: rcmgr.BaseLimit{
			Memory: int64(maxMemory / 4), // Give a better limitation that is related to the max memory param instead of a fixed default value coming from libp2p defaults.
			FD:     int(numFD / 4),

			Conns:         bigEnough,
			ConnsInbound:  int((maxMemory / 1024) / 4), // Follow the same logic as in System limits
			ConnsOutbound: bigEnough,

			Streams:         bigEnough,
			StreamsInbound:  bigEnough,
			StreamsOutbound: bigEnough,
		},
```

I think we need to change transient limits too to also increase these limits if we have enough resources. I think 1/4 of the total available resources by default is ok, but we can change that if you think we should give more or less to transient limits.